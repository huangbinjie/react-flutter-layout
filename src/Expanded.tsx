import React, { PropsWithChildren } from 'react'

export default function Expanded({ flex = 1, children }: PropsWithChildren<{ flex?: number }>) {
  return <div style={{ flex }}>{children}</div>
}
