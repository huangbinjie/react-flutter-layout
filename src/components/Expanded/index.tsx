import React from 'react'
import { style as jss } from 'typestyle'
import classnames from 'classnames'
import { only } from '../../utils'

export type ExpandedProps = {
  className?: string
  flex?: number
  children?: React.ReactNode
  style?: React.CSSProperties
}

export default function Expanded({ className = '', flex = 1, children, ...restProps }: ExpandedProps) {
  const classname = jss({
    flex: 1,
    // https://stackoverflow.com/a/68666833
    // 修复 Expand 可能因为内容导致宽度不对的问题, 强制宽度优先采用 flex-grow 给予的宽度
    // 比如 Row 下面有2 Expand，可能因为内容过长导致没有被平均分配
    minWidth: 0
  })

  return (
    <div debug-label="Expanded" className={classnames(classname, className)} {...restProps}>
      {only(children)}
    </div>
  )
}
