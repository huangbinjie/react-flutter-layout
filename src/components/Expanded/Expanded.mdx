---
name: Expanded
menu: components
---

import { Playground, Props } from 'docz'
import Container from '../Container'
import Row from '../Row'
import Center from '../Center'
import Expanded from './'

## Expanded

一个会自动占满 `Row`, `Column`可用空间的组件。等价于 `flex: 1`

### Basic

<Playground>
  <Container width="100%">
    <Row>
      <Container width="100px" height="100px" color="#ececec">
        <Center>居中</Center>
      </Container>
      <Expanded style={{background: "coral"}}>
        <Center>
          <Container width="100px" height="100px" color="blue">
            <Center style={{color: "#fff"}}>居中</Center>
          </Container>
        </Center>
      </Expanded>
    </Row>
  </Container>
</Playground>

## Props

<Props of={Expanded} />
