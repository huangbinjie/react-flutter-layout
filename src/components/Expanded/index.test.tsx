import React from 'react'
import renderer from 'react-test-renderer'

import Expanded from '.'

describe('Expanded', () => {
  test('basic', () => {
    const tree = renderer.create(
      <Expanded>empty content</Expanded>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
