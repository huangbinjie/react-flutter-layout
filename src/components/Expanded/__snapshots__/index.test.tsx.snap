// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expanded basic 1`] = `
<div
  className="f1n4s149"
  debug-label="Expanded"
>
  empty content
</div>
`;
